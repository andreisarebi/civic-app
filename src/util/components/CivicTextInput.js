import React from 'react';
import { StyleSheet, TextInput } from 'react-native';
import PropTypes from 'prop-types';
import colors from '../../styles/colors';

class CivicTextInput extends React.Component {
  static propTypes = {
    // TODO: list all other TextInput props
    onBlur: PropTypes.func,
    onFocus: PropTypes.func,
  };

  static defaultProps = {
    onBlur: undefined,
    onFocus: undefined,
  };

  constructor(props) {
    super(props);
    this.state = { borderBottomColor: colors.lightGray2 };
  }

  onBlur = () => {
    if (this.props.onBlur) {
      this.props.onBlur();
    }
    this.setState({ borderBottomColor: colors.lightGray2 });
  };

  onFocus = () => {
    if (this.props.onFocus) {
      this.props.onFocus();
    }
    this.setState({ borderBottomColor: colors.darkBlue });
  };

  render() {
    return (
      <TextInput
        style={[styles.textInput, { borderBottomColor: this.state.borderBottomColor }]}
        onBlur={this.onBlur}
        onFocus={this.onFocus}
        {...this.props}
      />
    );
  }
}

const styles = StyleSheet.create({
  textInput: {
    color: colors.black,
    fontSize: 18,
    paddingTop: 10,
    paddingBottom: 10,
  },
});

export default CivicTextInput;
