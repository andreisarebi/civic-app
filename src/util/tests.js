import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';
import store from '../../App/configureStore';

export const createSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Component {...props} />);
  return renderer.getRenderOutput();
};

export const createConnectedSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Provider store={store}><Component {...props} /></Provider>);
  return renderer.getRenderOutput();
};
