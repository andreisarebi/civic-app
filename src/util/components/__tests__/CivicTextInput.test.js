import { createSnapshot } from '../../tests';
import CivicTextInput from '../CivicTextInput';

describe('CivicTextInput', () => {
  it('renders correctly', () => {
    const snapshot = createSnapshot(CivicTextInput);
    expect(snapshot).toMatchSnapshot();
  });
});
