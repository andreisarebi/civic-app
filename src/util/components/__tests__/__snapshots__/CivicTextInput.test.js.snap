// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CivicTextInput renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  onBlur={undefined}
  onFocus={undefined}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 18,
        "paddingBottom": 10,
        "paddingTop": 10,
      },
      Object {
        "borderBottomColor": "#e0e0e0",
      },
    ]
  }
/>
`;
